raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -243,22 +244,27 @@ contract TokenNetwork is Utils {
 
     /// @notice Sets the channel participant total deposit value.
     /// Can be called by anyone.
+    /// @param channel_identifier Identifier for the channel on which this operation takes place.
     /// @param participant Channel participant whose deposit is being set.
     /// @param total_deposit The total amount of tokens that the participant will have
     /// as a deposit.
     /// @param partner Channel partner address, needed to compute the channel identifier.
-    function setTotalDeposit(address participant, uint256 total_deposit, address partner)
-        isOpen(participant, partner)
+    function setTotalDeposit(
+        uint256 channel_identifier,
+        address participant,
```

Comment:
So as Augusto also [commented](https://github.com/raiden-network/raiden-contracts/pull/198/files#r206564742) I find it a bit weird that we have this interface now.

I had assumed that with the introduction of the channel identifier we would no longer need the `address participant` in the function call interface for the contracts and thought that just passing `channel_identifier` would be enough.

Looking at the contract I see that the [Participant](https://github.com/raiden-network/raiden-contracts/blob/master/raiden_contracts/contracts/TokenNetwork.sol#L48) struct does not have the address and thus I guess it's an attempt to optimize for gas by not saving the participant address in the state and instead giving it as argument?

Is there any other reason that is blocking us from using only the single `channel_identifier` or could deoptimizing for gas for this case allow for a cleaner interface?

Reply:
> I had assumed that with the introduction of the channel identifier we would no longer need the address participant in the function call interface for the contracts and thought that just passing channel_identifier would be enough.

My bad. We still need the addresses.

>  attempt to optimize for gas by not saving the participant address in the state and instead giving it as argument?

Yes, gas optimization was always a requirement as far as I know, unless it was about over-optimization (e.g. being concerned about `uint248` vs. `uint256` while in development). In the February retreat we were actually discussing having minimal storage (hashes) and giving the values to each function. The current state is a good compromise I think.

Also, this gas optimization is about data structure. If you change the contract's data structure, you need to refactor everything.

I would personally leave the mapping as it is.

Reply:
> Yes, gas optimization was always a requirement as far as I know, unless it was about over-optimization

You should optimize for a bottleneck that has been identified but not base the entire architecture around it.

>  you need to refactor everything.

In the contracts only, right?

> I would personally leave the mapping as it is

So just tell me, if you add the participant addresses in the storage can we have a cleaner contract interface for the cost of 2 SSTORE or is it more complicated than that and there is another reason apart from gas optimization?

Reply:
I can think of at least one code complication with storing the participant addresses. You would have to add something like this every time you would need the other address or you need to differentiate between them:

```
function (channel_identifier, participant) {
    address partner;
    if(participant == storage.participant1_address) {
        partner = storage.participant2_address
    } else {
        partner = storage.participant1_address
    }
}
```
Something like the above would be needed in `setTotalDeposit`, `setTotalWithdraw`, `closeChannel`, `updateNonClosingBalanceProof`, `settleChannel` etc.

In my opinion, complicated code is worse than an interface with more arguments.

I remember taking things like this into consideration when I made the decision to implement the participant data as a mapping. Agreed, I should have documented it in writing better.

Also, for the record, this has not been changed since the first commit in this repo: https://github.com/raiden-network/raiden-contracts/commit/921c006a8fac5e556e238808667b48016b4c2bab#diff-1bdf2ff343918b98864fc26df16cc1ebR65

Reply:
> Agreed, I should have documented it in writing better so I would not need to re-think this again now.

I never mentioned anything about lack of documentation from your side.

> Also, for the record, this has not been changed since the first commit in this repo: 

That's irrelevant.

> Something like the above would be needed 

I see. Yes that is true, you would need to differentiate between the addresses to figure our which participant is who.

I don't think that such an addition would make the code more complicated and I am sure you would find a nice way to optimize it.

---

But after reading again what I wrote and putting more thought into it I will no longer insist on this. My thinking is the following:

- You, loredana, have put a lot of thought into this, know this code well and your recommendation is it should stay as is.
- 40,000 gas less is not too much, but it's not an insigificant amount either.
- Our priority atm is to go to mainnet asap and refactoring this would add more work, thus take us more time to implement, test, pr review.
- The cleaner interface argument I described [above](https://github.com/raiden-network/raiden-contracts/pull/198#discussion_r206922595) sounded really important to me back then, especially as this was one of my expectations(the other being cleaner channel lifecycle differentiation through the channel identifier) from this PR. But in reality the only consumer of this interface right now is the Raiden client. In the Raiden client we got all the info so data availability is not an obstacle. We are gonna change the proxies anyway so I thought going for a clean interface is the best option but compared to all the other negatives I described above the interface argument sounds more like code purity than practicality which is not what we strive for.

Reply:
Ok.

> especially as this was one of my expectations

Yes, I also contributed to that with my mention from https://github.com/raiden-network/raiden-contracts/issues/193:
> Currently there is a lot of confusion when calling functions because we need to provide both participants, but usually one of the participants also has a role and the caller needs to pay attention at the participant addresses order. This has brought even myself a lot of pain when testing.

Added an update there. As I said, my bad.